### DISCUSSION TITLE
Feature request - customisable Username divider symbol

### DISCUSSION THREAD

**Discussion by VeselaHouba** [postalserver/postal#1498] [Feature suggestions]
Would it be possible to define which symbol will be used for SMTP username generating?

Currently the username is generated as
`organisation/server`
E.g.
![image](https://user-images.githubusercontent.com/6410423/105015596-013c4300-5a42-11eb-86ec-33991dc614c5.png)

But the slash `/` is not acceptable as username on many systems.
![image](https://user-images.githubusercontent.com/6410423/105015737-25981f80-5a42-11eb-96c7-7fa7a0025741.png)

So it would be nice to have option to define the divider, leaving the slash as default to avoid breaking changes during upgrades.

PS: I'm not saying it's postal's fault, that another system doesn't support special characters.

**Comment by willpower232** [comment]
I can see that this would be useful however having recently had a look in https://github.com/postalhq/postal/blob/master/lib/postal/smtp_server/client.rb, I noticed that the username field is actually not used currently so you can put whatever value you like in your client softwares username field.

Out of curiosity, what software are you using that says the special characters are invalid?

**Comment by VeselaHouba** [comment]
Oh my, if only I knew ...
The message above was from Unifi NVR and I'm pretty sure I've seen it somewhere before, but can't remember where.
I confirm that any username works. It would be nice to mention it somewhere in docu :)

Marking as closed, as the feature is not needed.

**Comment by willpower232** [comment]
I'll actually leave this open because I believe this behaviour needs documenting or correcting.

**Comment by lcx** [comment]
Other systems that complaint about / would be printers. For example xerox has this issue and I know I came across some others which have the same issues. I never bothered to look in the code as I really assumed that user is required. 🤦‍♂️ I should have known better since using api also requires no username.

**Comment by r3nor** [comment]
Is this still an ongoing issue? Can I really put whatever in the username field?

**Reply by willpower232** [reply]
yes, postal only uses the password/api key to verify identity but also it will verify that the sending domain matches

**Comment by adamcooke** [comment]
Just to clear up some confusion on this issue for anyone finding it, the username doesn't matter if the `LOGIN` or `PLAIN` authentication mechanisms are used. If you use `CRAM-MD5` the username does matter and should be provided in the format of `org/server`. You can use also a `_` in place of the `/` if needed.

**Reply by willpower232** [reply]
is it worth having an SMTP Auth For Newbies page in the docs?

**Reply by adamcooke** [reply]
You mean, like this one? https://docs.postalserver.io/features/smtp-authentication 😆 I put that up last night after this.

**Reply by willpower232** [reply]
Haha magic!
